homework/list.js

```diff
@@ -47,33 +40,39 @@
       li.dataset.noteId = i;
       li.classList.add('note-title');
       li.textContent = note.title;
-      // Note: buff is captured, so we now have a
-      // little closure naturally.
       buff.appendChild(li);
     });
     ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
+    this._wrapper.appendChild(ul);
+  },
 
-  function fetchList(afterFetch) {
+  fetchList(afterFetch) {
+	  return new Promise((function(resolve,reject){
     var xhr = new XMLHttpRequest();
     xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
     xhr.responseType = 'json';
     xhr.onreadystatechange = function(e) {
-      // Watch out: we have a mysterious unknown 'this'.
       if (this.readyState === 4 && this.status === 200) {
         var listData = this.response;
-        // The flow ends here.
-        afterFetch(listData);
+        resolve(listData);
       } else if (this.status !== 200 ){
-        // Ignore error in this case.
+        reject(new Error(xhr.statusText));
       }
-    };
+    }.bind(xhr);
     xhr.send();
-  }
-
-  document.addEventListener('DOMContentLoaded', function(event) {
-    start();
-  });
+  }))},
 
-})();
+  start() {
+	this._wrapper = document.querySelector('#note-list-wrapper')
+    this.fetchList(function(data) {
```

Comment:
**WRONG**: since you have "Promisifed" the method, you don't need to use the callback. And from this code I bet you didn't test your code by executing it, since it apparently won't work.

Remember: you either use the callback to perform the following steps (the way I require you to refactor), or change it to Promise style and chain the actions with `then` and `catch`. To leave code like this only shows you don't really understand what I've taught you, and you should ask more questions before you submit this. **So I strongly suggest you correct you homework as soon as possible, and ask me for help if you still don't really understand how to use Promise. Otherwise, you may get into trouble**.